src/Type/Introspection.php

```diff
@@ -39,7 +39,7 @@ class Introspection
     const TYPE_FIELD_NAME      = '__type';
     const TYPE_NAME_FIELD_NAME = '__typename';
 
-    /** @var Type[] */
+    /** @var array<string, mixed> */
```

Comment:
```suggestion
    /** @var array<string, Type> */
```

Reply:
Can't do this one, unfortunately--$map also stores `FieldDefinition` (which seems odd)

Reply:
What about in the meanwhile then?

```suggestion
    /** @var array<string, FieldDefinition|Type> */
```

Reply:
Oh, that's a good idea! I'll try that when I get home...

Reply:
Done, thanks!

Reply:
Had to revert it--Scrutinizer (rightfully) complains. I'll sort this out in more depth in future phpstan levels.